### PR TITLE
New "Join this colony" button

### LINKED ISSUE #3383: New Join this Colony button
### Description

Add a new `Join this Colony` button.

### Design

[Figma link](https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=6736%3A90572)

Screenshot 2022-05-10 194548
Screenshot 2022-05-10 194601

### PR DESCRIPTION
**Pull request by arrenv**
## Description

Replaces `Join` button with newly designed `Join this colony` button.

Note: After confirming with Karol, the left alignment of the button is meant to sit to the left, so the hover state of the address and button line up. This is slightly different to the provided spec.

**New stuff** ✨

* Add new button styling

252f263d-5462-4d96-8bb9-b19159f18623

**Changes** 🏗

* Refactored the `ColonyTitle.tsx` and `ColonySubscription.tsx` to allow for both the new button position and the inline colony kebab menu.

**Deletions** ⚰️

* Removed some now redundant styling.

Resolves #3383

### CODE REVIEW

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css [hunk 1]
```diff
@@ -1,19 +1,36 @@
 .main {
   composes: flexContainerRow flexAlignCenter from '~styles/layout.css';
-  margin-left: 2px;
+  margin-top: -4px;
```
Comment (chinins) [inline src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css:3]: I really dislike negative margin. Why do we need it here? Can't we fix it any other way?

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx [hunk 2]
```diff
@@ -68,6 +74,8 @@ const ColonySubscription = ({
 
   const isNetworkAllowed = checkIfNetworkIsAllowed(networkId);
 
+  const colonyMenu = isSubscribed ? styles.colonySubscribed : '';
```
Comment (chinins) [inline src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.tsx:77]: It's a nitpick but I think it's more readable and clear when it's inline for className...

**Review by chinins — APPROVED** [review]
Very nice!

It's a separate issue but I've been thinking about it for a while so I'll just mention it here.

As a user it always bothered me that I have to click the button twice - once to create the username and second time to actually join the colony. I think it might be confusing for users who might not realise it straight away or they might forget and expect to already have joined. Not sure that we can/should do something about it right now.

It is definitely more clear right now with this new button...

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css [hunk 3]
```diff
@@ -51,13 +68,24 @@
   left: 0;
 }
 
-.createUserRedirect {
-  font-size: var(--size-tiny);
+.joinColonyBtn {
```
Comment (danbr) [inline src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css:71]: nitpicky... but.... for the naming convention, until now the names started with 'colony' (i.e. colonySubscribed, colonyJoin) but here you switch to joinColony(Btn).

Same in the .tsx file.

**Review by danbr — APPROVED** [review]
LGTM nice job. To me, the text doesn't quite 'look' vertically centred on the button, despite inspecting the element. Off by a few px's, perhaps it's the font creating an optical illusion.

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css (hunk 1 above)
Comment (rdig) [reply src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css:3]: Please leave a note if using negative margins explaining why

**Review by rdig — APPROVED** [review]
Pretty much ok

**Review by ArmandoGraterol — APPROVED** [review]
Nice work! I only noticed that the `border-radius` should be doubled as currently it's `4px` but in the specs is `8px`. Is that intentional?

Implementation
FireShot Capture 594 - Actions - Colony - wonker - localhost

Specs
FireShot Capture 593 - Colony Simplified – Figma - www figma com


Either way, I'm approving.

**Comment by arrenv** [comment]
> @chinins 
> As a user it always bothered me that I have to click the button twice - once to create the username and second time to actually join the colony. I think it might be confusing for users who might not realise it straight away or they might forget and expect to already have joined. Not sure that we can/should do something about it right now.

This is actually something we were talking about in the Product Sync to improve onboarding, I will take a look at it. We also talked about passing a parameter in an invite link, like `http://localhost:9090/colony/cooldao/invite`? What are your thoughts on that?

File: src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (arrenv) [reply src/modules/dashboard/components/ColonyHome/ColonySubscription/ColonySubscription.css:3]: Surprise!! I have now removed it :)

**Comment by chinins** [comment]
> > @chinins
> > As a user it always bothered me that I have to click the button twice - once to create the username and second time to actually join the colony. I think it might be confusing for users who might not realise it straight away or they might forget and expect to already have joined. Not sure that we can/should do something about it right now.
> 
> This is actually something we were talking about in the Product Sync to improve onboarding, I will take a look at it. We also talked about passing a parameter in an invite link, like `http://localhost:9090/colony/cooldao/invite`? What are your thoughts on that?
> 
> EDIT: I decided to create a separate issue for this here - #3403

Very good!

**Pull request merged**